src/front/glsl/parser.rs

```diff
@@ -499,15 +505,15 @@ impl<'source, 'program, 'options> Parser<'source, 'program, 'options> {
             // TODO: Should we try to make constants here?
             // This is mostly a hack because we don't yet support adding
             // bodies to entry points for variable initialization
-            let maybe_constant = init.clone().and_then(|(root, meta)| {
-                self.program
-                    .solve_constant(ctx.ctx.expressions, root, meta)
-                    .ok()
-            });
+            let maybe_constant = init
+                .clone()
+                .and_then(|(root, meta)| self.program.solve_constant(ctx.ctx, root, meta).ok());
 
             let pointer = ctx.add_var(self.program, ty, name, maybe_constant, meta)?;
 
             if let Some((value, _)) = init {
+                ctx.ctx.emit_flush(ctx.body);
```

Comment:
this emit+start should probably be abstracted behind a helper, like `flush_expressions()`